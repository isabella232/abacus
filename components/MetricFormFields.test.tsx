/* eslint-disable @typescript-eslint/require-await */
import { act, fireEvent, screen } from '@testing-library/react'
import { Formik, FormikProps } from 'formik'
import MockDate from 'mockdate'
import React from 'react'

import { MetricFormData, metricToFormData } from '@/lib/form-data'
import Fixtures from '@/test-helpers/fixtures'
import { render } from '@/test-helpers/test-utils'

import MetricFormFields from './MetricFormFields'

MockDate.set('2020-07-21')

test('renders as expected for conversion metric', async () => {
  const metric = Fixtures.createMetricFull(1)
  const { container } = render(
    <Formik
      initialValues={{
        metric: metricToFormData(metric),
      }}
      onSubmit={
        /* istanbul ignore next; This is unused */
        () => undefined
      }
    >
      {(formikProps: FormikProps<{ metric: MetricFormData }>) => <MetricFormFields formikProps={formikProps} />}
    </Formik>,
  )

  expect(container).toMatchSnapshot()

  await act(async () => {
    fireEvent.click(screen.getByRole('radio', { name: 'Revenue' }))
  })
})

test('renders as expected for revenue metric', async () => {
  const metric = Fixtures.createMetricFull(2)
  const { container } = render(
    <Formik
      initialValues={{
        metric: metricToFormData(metric),
      }}
      onSubmit={
        /* istanbul ignore next; This is unused */
        () => undefined
      }
    >
      {(formikProps: FormikProps<{ metric: MetricFormData }>) => <MetricFormFields formikProps={formikProps} />}
    </Formik>,
  )

  expect(container).toMatchSnapshot()

  await act(async () => {
    fireEvent.click(screen.getByRole('radio', { name: 'Conversion' }))
  })
})

test('renders as expected for new metric', () => {
  const { container } = render(
    <Formik
      initialValues={{
        metric: metricToFormData({}),
      }}
      onSubmit={
        /* istanbul ignore next; This is unused */
        () => undefined
      }
    >
      {(formikProps: FormikProps<{ metric: MetricFormData }>) => <MetricFormFields formikProps={formikProps} />}
    </Formik>,
  )

  expect(container).toMatchSnapshot()
})
