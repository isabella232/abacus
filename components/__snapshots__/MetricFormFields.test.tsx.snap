// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected for conversion metric 1`] = `
<div>
  <div
    class="makeStyles-row-2"
  >
    <div
      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
        data-shrink="true"
        for="metric.name"
        id="metric.name-label"
      >
        Metric name
        <span
          aria-hidden="true"
          class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
      >
        <input
          aria-describedby="metric.name-helper-text"
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input"
          id="metric.name"
          name="metric.name"
          placeholder="metric_name"
          required=""
          type="text"
          value="metric_1"
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-5 PrivateNotchedOutline-legendNotched-6"
          >
            <span>
              Metric name
               *
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        class="MuiFormHelperText-root MuiFormHelperText-contained MuiFormHelperText-filled Mui-required"
        id="metric.name-helper-text"
      >
        Use snake_case.
      </p>
    </div>
  </div>
  <div
    class="makeStyles-row-2"
  >
    <div
      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
        data-shrink="true"
        for="metric.description"
        id="metric.description-label"
      >
        Metric description
        <span
          aria-hidden="true"
          class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
      >
        <textarea
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
          id="metric.description"
          name="metric.description"
          placeholder="Put your Metric description here!"
          required=""
          rows="4"
        >
          This is metric 1
        </textarea>
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-5 PrivateNotchedOutline-legendNotched-6"
          >
            <span>
              Metric description
               *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
  <div
    class="makeStyles-row-2"
  >
    <label
      class="MuiFormControlLabel-root"
    >
      <span
        class="MuiSwitch-root"
      >
        <span
          aria-disabled="false"
          aria-label="Higher is better"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-7 MuiSwitch-switchBase MuiSwitch-colorSecondary"
          label="Higher is better"
          variant="outlined"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              class="PrivateSwitchBase-input-10 MuiSwitch-input"
              id="metric.higherIsBetter"
              name="metric.higherIsBetter"
              type="checkbox"
              value="false"
            />
            <span
              class="MuiSwitch-thumb"
            />
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiSwitch-track"
        />
      </span>
      <span
        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        Higher is better
      </span>
    </label>
  </div>
  <div
    class="makeStyles-row-2"
  >
    <fieldset
      class="MuiFormControl-root"
    >
      <label
        class="MuiFormLabel-root Mui-required"
        id="metric-form-radio-metric-type-label"
      >
        Metric Type
        <span
          aria-hidden="true"
          class="MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        aria-labelledby="metric-form-radio-metric-type-label"
        class="MuiFormGroup-root"
        required=""
        role="radiogroup"
      >
        <label
          aria-label="Conversion"
          class="MuiFormControlLabel-root"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-7 MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-checked-8 Mui-checked MuiIconButton-colorSecondary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input-10"
                name="metric.parameterType"
                type="radio"
                value="conversion"
              />
              <div
                class="PrivateRadioButtonIcon-root-11 PrivateRadioButtonIcon-checked-13"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-12"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </div>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Conversion
          </span>
        </label>
        <label
          aria-label="Revenue"
          class="MuiFormControlLabel-root"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-7 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-10"
                name="metric.parameterType"
                type="radio"
                value="revenue"
              />
              <div
                class="PrivateRadioButtonIcon-root-11"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-12"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </div>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Revenue
          </span>
        </label>
      </div>
    </fieldset>
  </div>
  <div
    class="makeStyles-row-2"
  >
    <div
      class="makeStyles-root-14"
    >
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
          data-shrink="true"
          for="metric.eventParams"
          id="metric.eventParams-label"
        >
          Event Parameters
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
        >
          <textarea
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
            id="metric.eventParams"
            name="metric.eventParams"
            required=""
            rows="8"
          >
            [
  {
    "event": "event_name",
    "props": {
      "has_blocks": "true"
    }
  }
]
          </textarea>
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-5 PrivateNotchedOutline-legendNotched-6"
            >
              <span>
                Event Parameters
                 *
              </span>
            </legend>
          </fieldset>
        </div>
        
      </div>
    </div>
  </div>
  <div
    class="makeStyles-row-2"
  />
  <div
    class="makeStyles-root-15"
  >
    <p
      class="MuiTypography-root makeStyles-label-17 MuiTypography-body1"
    >
      Validation Errors
    </p>
    <p
      class="MuiTypography-root makeStyles-pre-16 MuiTypography-body1"
    >
      {}
    </p>
  </div>
</div>
`;

exports[`renders as expected for new metric 1`] = `
<div>
  <div
    class="makeStyles-row-36"
  >
    <div
      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-required Mui-required"
        data-shrink="true"
        for="metric.name"
        id="metric.name-label"
      >
        Metric name
        <span
          aria-hidden="true"
          class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
      >
        <input
          aria-describedby="metric.name-helper-text"
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input"
          id="metric.name"
          name="metric.name"
          placeholder="metric_name"
          required=""
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-37 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-39 PrivateNotchedOutline-legendNotched-40"
          >
            <span>
              Metric name
               *
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        class="MuiFormHelperText-root MuiFormHelperText-contained Mui-required"
        id="metric.name-helper-text"
      >
        Use snake_case.
      </p>
    </div>
  </div>
  <div
    class="makeStyles-row-36"
  >
    <div
      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-required Mui-required"
        data-shrink="true"
        for="metric.description"
        id="metric.description-label"
      >
        Metric description
        <span
          aria-hidden="true"
          class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
      >
        <textarea
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
          id="metric.description"
          name="metric.description"
          placeholder="Put your Metric description here!"
          required=""
          rows="4"
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-37 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-39 PrivateNotchedOutline-legendNotched-40"
          >
            <span>
              Metric description
               *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
  <div
    class="makeStyles-row-36"
  >
    <label
      class="MuiFormControlLabel-root"
    >
      <span
        class="MuiSwitch-root"
      >
        <span
          aria-disabled="false"
          aria-label="Higher is better"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-41 MuiSwitch-switchBase MuiSwitch-colorSecondary PrivateSwitchBase-checked-42 Mui-checked"
          label="Higher is better"
          variant="outlined"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input-44 MuiSwitch-input"
              id="metric.higherIsBetter"
              name="metric.higherIsBetter"
              type="checkbox"
              value="true"
            />
            <span
              class="MuiSwitch-thumb"
            />
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiSwitch-track"
        />
      </span>
      <span
        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        Higher is better
      </span>
    </label>
  </div>
  <div
    class="makeStyles-row-36"
  >
    <fieldset
      class="MuiFormControl-root"
    >
      <label
        class="MuiFormLabel-root Mui-required"
        id="metric-form-radio-metric-type-label"
      >
        Metric Type
        <span
          aria-hidden="true"
          class="MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        aria-labelledby="metric-form-radio-metric-type-label"
        class="MuiFormGroup-root"
        required=""
        role="radiogroup"
      >
        <label
          aria-label="Conversion"
          class="MuiFormControlLabel-root"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-41 MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-checked-42 Mui-checked MuiIconButton-colorSecondary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input-44"
                name="metric.parameterType"
                type="radio"
                value="conversion"
              />
              <div
                class="PrivateRadioButtonIcon-root-45 PrivateRadioButtonIcon-checked-47"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-46"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </div>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Conversion
          </span>
        </label>
        <label
          aria-label="Revenue"
          class="MuiFormControlLabel-root"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-41 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-44"
                name="metric.parameterType"
                type="radio"
                value="revenue"
              />
              <div
                class="PrivateRadioButtonIcon-root-45"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-46"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </div>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Revenue
          </span>
        </label>
      </div>
    </fieldset>
  </div>
  <div
    class="makeStyles-row-36"
  >
    <div
      class="makeStyles-root-48"
    >
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink="true"
          for="metric.eventParams"
          id="metric.eventParams-label"
        >
          Event Parameters
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
        >
          <textarea
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
            id="metric.eventParams"
            name="metric.eventParams"
            required=""
            rows="8"
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-37 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-39 PrivateNotchedOutline-legendNotched-40"
            >
              <span>
                Event Parameters
                 *
              </span>
            </legend>
          </fieldset>
        </div>
        
      </div>
    </div>
  </div>
  <div
    class="makeStyles-row-36"
  />
  <div
    class="makeStyles-root-49"
  >
    <p
      class="MuiTypography-root makeStyles-label-51 MuiTypography-body1"
    >
      Validation Errors
    </p>
    <p
      class="MuiTypography-root makeStyles-pre-50 MuiTypography-body1"
    >
      {}
    </p>
  </div>
</div>
`;

exports[`renders as expected for revenue metric 1`] = `
<div>
  <div
    class="makeStyles-row-19"
  >
    <div
      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
        data-shrink="true"
        for="metric.name"
        id="metric.name-label"
      >
        Metric name
        <span
          aria-hidden="true"
          class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
      >
        <input
          aria-describedby="metric.name-helper-text"
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input"
          id="metric.name"
          name="metric.name"
          placeholder="metric_name"
          required=""
          type="text"
          value="metric_2"
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-20 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-22 PrivateNotchedOutline-legendNotched-23"
          >
            <span>
              Metric name
               *
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        class="MuiFormHelperText-root MuiFormHelperText-contained MuiFormHelperText-filled Mui-required"
        id="metric.name-helper-text"
      >
        Use snake_case.
      </p>
    </div>
  </div>
  <div
    class="makeStyles-row-19"
  >
    <div
      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
        data-shrink="true"
        for="metric.description"
        id="metric.description-label"
      >
        Metric description
        <span
          aria-hidden="true"
          class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
      >
        <textarea
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
          id="metric.description"
          name="metric.description"
          placeholder="Put your Metric description here!"
          required=""
          rows="4"
        >
          This is metric 2
        </textarea>
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-20 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-22 PrivateNotchedOutline-legendNotched-23"
          >
            <span>
              Metric description
               *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
  <div
    class="makeStyles-row-19"
  >
    <label
      class="MuiFormControlLabel-root"
    >
      <span
        class="MuiSwitch-root"
      >
        <span
          aria-disabled="false"
          aria-label="Higher is better"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-24 MuiSwitch-switchBase MuiSwitch-colorSecondary"
          label="Higher is better"
          variant="outlined"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              class="PrivateSwitchBase-input-27 MuiSwitch-input"
              id="metric.higherIsBetter"
              name="metric.higherIsBetter"
              type="checkbox"
              value="false"
            />
            <span
              class="MuiSwitch-thumb"
            />
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiSwitch-track"
        />
      </span>
      <span
        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        Higher is better
      </span>
    </label>
  </div>
  <div
    class="makeStyles-row-19"
  >
    <fieldset
      class="MuiFormControl-root"
    >
      <label
        class="MuiFormLabel-root Mui-required"
        id="metric-form-radio-metric-type-label"
      >
        Metric Type
        <span
          aria-hidden="true"
          class="MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        aria-labelledby="metric-form-radio-metric-type-label"
        class="MuiFormGroup-root"
        required=""
        role="radiogroup"
      >
        <label
          aria-label="Conversion"
          class="MuiFormControlLabel-root"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-24 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-27"
                name="metric.parameterType"
                type="radio"
                value="conversion"
              />
              <div
                class="PrivateRadioButtonIcon-root-28"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-29"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </div>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Conversion
          </span>
        </label>
        <label
          aria-label="Revenue"
          class="MuiFormControlLabel-root"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-24 MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-checked-25 Mui-checked MuiIconButton-colorSecondary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input-27"
                name="metric.parameterType"
                type="radio"
                value="revenue"
              />
              <div
                class="PrivateRadioButtonIcon-root-28 PrivateRadioButtonIcon-checked-30"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-29"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </div>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Revenue
          </span>
        </label>
      </div>
    </fieldset>
  </div>
  <div
    class="makeStyles-row-19"
  />
  <div
    class="makeStyles-row-19"
  >
    <div
      class="makeStyles-root-31"
    >
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
          data-shrink="true"
          for="metric.revenueParams"
          id="metric.revenueParams-label"
        >
          Revenue Parameters
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
        >
          <textarea
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
            id="metric.revenueParams"
            name="metric.revenueParams"
            required=""
            rows="8"
          >
            {
  "refundDays": 4,
  "productSlugs": [
    "xx-bundles"
  ],
  "transactionTypes": [
    "new purchase"
  ]
}
          </textarea>
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-20 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-22 PrivateNotchedOutline-legendNotched-23"
            >
              <span>
                Revenue Parameters
                 *
              </span>
            </legend>
          </fieldset>
        </div>
        
      </div>
    </div>
  </div>
  <div
    class="makeStyles-root-32"
  >
    <p
      class="MuiTypography-root makeStyles-label-34 MuiTypography-body1"
    >
      Validation Errors
    </p>
    <p
      class="MuiTypography-root makeStyles-pre-33 MuiTypography-body1"
    >
      {}
    </p>
  </div>
</div>
`;
